src/containers/Plot/index.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from "react";
+import { connect } from "react-redux";
+import PlotlyPlot from "react-plotly.js";
+// import { ActionCreator } from "redux";
+
+import { State } from "../../state/types";
+import { PlotConfig } from "../ResultsPanel/types";
+import { getCurrentTime } from "../../state/selection/selectors";
+// import { changeTime } from "../../state/selection/actions";
+// import { ChangeTimeAction } from "../../state/selection/types";
+import { PLOT_STYLE } from "../ResultsPanel/constants";
+
+interface PlotProps {
+    plotConfig: PlotConfig;
+    time: number;
+    // changeTime: ActionCreator<ChangeTimeAction>;
+}
+
+const styles = require("./style.css");
+
+class Plot extends React.Component<PlotProps, {}> {
+    shouldComponentUpdate() {
+        return this.props.plotConfig.hasTimeIndicator;
+    }
+
+    public render(): JSX.Element | null {
+        const { plotConfig, time } = this.props;
+        const { hasTimeIndicator, data, layout } = plotConfig;
+        const TIME_INDICATOR_LINE = "timeIndicatorLine";
+
+        if (hasTimeIndicator && time !== 0) {
+            const lastPlot = data[data.length - 1];
+            // Update the time for the time indicator line if it already exists,
+            // otherwise add a time indicator line
+            if (lastPlot.name === TIME_INDICATOR_LINE) {
```

Comment:
For clarity I would probably functionalize this into methods 
```findTimeIndicatorPlot() => find(this.props.data), {name : TIME_INDICATOR_LINE}```

 that just checks if there is a plot named `TIME_INDICATOR_LINE` and returns it (the getting the last plot would work fine here, and is faster, so you can do that too, I just think this is an clearer check ). 

Another one that is `updateTimeIndicatorPosition(timePlot) => timePlot.x[this.props.time, this.props.time]` that updates the plot, 

and a `addTimeIndicator()` that does lines 38-53

Reply:
I hope the shifting around of code and some improved variable names / comments in my latest push make things more clear and address this comment!